SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -1717,10 +1714,10 @@ void CEulerSolver::Set_MPI_ActDisk(CSolver **solver_container, CGeometry *geomet
           if ((iDomain == jDomain) && (geometry->node[iPoint]->GetDomain())) {
             
             for (iVar = 0; iVar < nPrimVar; iVar++) {
-              Buffer_Send_PrimVar[(nPrimVar_)*(PointTotal_Counter+iPointTotal)+iVar] = node[iPoint]->GetPrimitive(iVar);
+              Buffer_Send_PrimVar[(nPrimVar_)*(PointTotal_Counter+iPointTotal)+iVar] = snode->GetPrimitive(iPoint,iVar);
             }
             if (rans) {
-              Buffer_Send_PrimVar[(nPrimVar_)*(PointTotal_Counter+iPointTotal)+nPrimVar] = solver_container[TURB_SOL]->node[iPoint]->GetSolution(0);
+              Buffer_Send_PrimVar[(nPrimVar_)*(PointTotal_Counter+iPointTotal)+nPrimVar] = solver_container[TURB_SOL]->node->GetSolution(iPoint,0);
               Buffer_Send_PrimVar[(nPrimVar_)*(PointTotal_Counter+iPointTotal)+(nPrimVar+1)] = 0.0;
             }
```

Comment:
Just so I understand how this node - snode relationship works:

snode is used only by the solver that it is instantiated in. It points to the node variable in that class, but is statically cast to be specific to the solver? And this is done to remove the need for virtual functions?

The fact that the node and snode pointers are the same means that if accessing the node of a different class, you can still just use the node variable?

Just trying to wrap my head around this

Reply:
You can access node but not snode from the outside.
The main purpose was not to remove the need for virtual functions, it was to allow `final` virtual functions to be inlined, without derived type information the compiler needs to insert all the dynamic dispatch mechanisms which are relatively expensive when all you are doing is getting a double. Close to the bottom of #716 this is explained in more detail.
As an added bonus you can indeed add data members to a derived CVariable and not to the base (i.e. no virtual) provided you only access through snode.
As clever as the strategy may look it is not ideal design, ideally derived solvers would have a node of the right type for internal use, and for access from the outside they would cast to base class and return a CVariable pointer (less confusing than having two variables that allow you to do the "same").

Reply:
Makes sense. Would that change be in this PR or after? And that's what is referred to in your comment:

> I want to be done with this PR folks... This is really tiring to maintain so if you don't want it please just tell me...
> Can we agree on:
> Instead of CSolver having the `node` field it will have a pure virtual function `CVariable* GetNodes()`, derived solvers need to implement this method and have a `nodes` field of the most appropriate type (e.g. CEulerVariable for CEulerSolver).
> 
> I know some of you do not like the name `nodes` but there is something to be said about consistency (that has always been the name) and I do not think changing a generic name for another generic name justifies breaking the formatting everywhere and having needlessly long calls to get some data.

For what its worth, I think I agree that the name `nodes` is the best bet.